### ISSUE TITLE
[CT-3310] Support `--empty` flag for dbt seed

### ISSUE THREAD

**Issue by graciegoheen** [dbt-labs/dbt-core#8981]
### Housekeeping

- [X] I am a maintainer of dbt-core

### Short description

`dbt seed --empty` produces empty (0 row) seeds in the warehouse:
- on most DWHs this would be quite simple, since we already create empty table + then insert
- BQ implementation is quite different though

### Acceptance criteria

`dbt seed --empty` produces empty (0 row) seeds in the warehouse

### Impact to Other Teams

impact adapters

### Will backports be required?

no

### Context

blocked by https://github.com/dbt-labs/dbt-core/issues/8980

**Comment by jtcohen6** [comment]
I'm moving issues from `dbt-core` to `dbt-adapters` that relate to implementation logic which now lives in the latter, but I'm leaving this one here because it's possible we could implement this within `dbt-core`

**Comment by dbeatty10** [comment]
### Implementation hint

See https://github.com/dbt-labs/dbt-core/pull/10962 for how this was implemented for snapshots.

**Comment by ChenyuLInx** [comment]
Potentially, put a limit [here](https://github.com/dbt-labs/dbt-core/blob/03b621f390a24a0de3fcd3e3ef98acabbb96ea06/core/dbt/task/seed.py#L33)

**Comment by dbeatty10** [comment]
> Potentially, put a limit [here](https://github.com/dbt-labs/dbt-core/blob/03b621f390a24a0de3fcd3e3ef98acabbb96ea06/core/dbt/task/seed.py#L33)

Good idea! 💡

When `empty` is `True` for a seed, then we could try something like [this](https://agate.readthedocs.io/en/latest/api/table.html#agate.Table.limit):

```python
        result.agate_table = agate_result.table.limit(0)
```

And otherwise leave it as-is:

```python
        result.agate_table = agate_result.table
```

**Comment by graciegoheen** [comment]
Would that create a table in the warehouse with the correct data types for the columns?

**Comment by FishtownBuildBot** [comment]
Opened a new issue in dbt-labs/docs.getdbt.com: https://github.com/dbt-labs/docs.getdbt.com/issues/8996
